### PR TITLE
[SIL] Add UTF-8 string `SymbolicValue`.

### PR DESCRIPTION
**Pull request by dan-zheng**
Add a UTF-8 string case to `SymbolicValue` and support it as a `graph_op`
attribute value.

Also do some other NFC cleanup.

### CODE REVIEW

File: include/swift/SIL/SILConstants.h [hunk 1]
```diff
@@ -239,6 +248,11 @@ class SymbolicValue {
 
   APFloat getFloatValue() const;
 
+  static SymbolicValue getString(const StringRef &string,
```
Comment (lattner) [inline include/swift/SIL/SILConstants.h]: Please pass the StringRef by value, it is a small POD type.  Please add a comment that the bytes are in UTF8 to both methods.

File: lib/ParseSIL/ParseSIL.cpp [hunk 2]
```diff
@@ -997,12 +998,10 @@ static bool parseSymbolicValue(SymbolicValue &value, SILParser &SP,
...
   if (P.Tok.is(tok::string_literal)) {
-    // TODO: Uncomment when `getStringValue` is implemented.
-    // StringRef rawString = P.Tok.getText().drop_front().drop_back();
-    // value = SymbolicValue::getStringValue(rawString, allocator);
-    // return false;
-    P.diagnose(P.Tok, diag::sil_graph_op_unhandled_attr_value);
-    return true;
+    StringRef rawString = P.Tok.getText().drop_front().drop_back();
+    value = SymbolicValue::getString(rawString, allocator);
+    P.consumeToken();
```
Comment (lattner) [inline lib/ParseSIL/ParseSIL.cpp]: Please pass tok::string_literal to consumeToken (to follow convention)

File: lib/SIL/SILConstants.cpp [hunk 3]
```diff
@@ -239,31 +242,94 @@ SymbolicValue SymbolicValue::getFloat(const APFloat &value,
...
+/// This is a representation of an UTF-8 string, stored as a trailing array of
+/// bytes.  Elements of this value are bump-pointer allocated.
+struct alignas(uint64_t) StringSymbolicValue final
+  : private llvm::TrailingObjects<StringSymbolicValue, char> {
+    friend class llvm::TrailingObjects<StringSymbolicValue, char>;
+
+  /// The number of bytes in the trailing array.
+  const unsigned numBytes;
+
+  static StringSymbolicValue *create(const StringRef &string,
```
Comment (lattner) [inline lib/SIL/SILConstants.cpp]: StringRef by value plz.

**Review by lattner — APPROVED** [review]
Totally awesome, great work Dan!

File: include/swift/SIL/SILConstants.h (hunk 1 above)
Comment (dan-zheng) [reply include/swift/SIL/SILConstants.h]: Thanks, good catch! Done in https://github.com/apple/swift/pull/17235/commits/013f7a48819b2daee4a4f09e953fb7e33ccb96cd.

File: lib/ParseSIL/ParseSIL.cpp (hunk 2 above)
Comment (dan-zheng) [reply lib/ParseSIL/ParseSIL.cpp]: Done.

File: lib/SIL/SILConstants.cpp (hunk 3 above)
Comment (dan-zheng) [reply lib/SIL/SILConstants.cpp]: Done.

**Comment by dan-zheng** [comment]
@swift-ci Please test tensorflow

**Comment by dan-zheng** [comment]
@swift-ci Please test tensorflow

**Comment by dan-zheng** [comment]
@swift-ci Please test tensorflow

**Comment by dan-zheng** [comment]
@swift-ci Please test tensorflow

**Pull request merged**